Automattic/wp-calypso
# A4A: Add Hosting section.

This pull request adds a hosting section to the Marketplace which includes skeleton pages for both Pressable and WPCOM.

<img width="1728" alt="Screenshot 2024-03-21 at 7 16 49 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/6e5a0db0-078f-46ed-bb92-3cb25d87f2cc">
<img width="1726" alt="Screenshot 2024-03-21 at 7 16 56 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/f493f254-fa3f-49c8-a873-148af5bcabab">
<img width="1728" alt="Screenshot 2024-03-21 at 7 17 03 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/80641d31-71b1-4052-a9c3-cc4845989629">

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/73

## Proposed Changes

* Transform current product listing components and hooks into shared code, including **Filter search** and **Listing section** and the **useProductAndPlans** hook.
* Update **useProductAndPlans** hook to include the **Pressable** filter.
* Implement a new hosting card that displays Hosting provider information in the main Hosting section.* Add **Pressable** and **WPCOM** overview pages' skeleton.

## Testing Instructions
* Use the live link and go to `/marketplace/hosting`
* Confirm that the Hosting section is rendered properly and loads the Pressable and Wordpress.com cards.
* Click the explore buttons and confirm it redirects to the correct Hosting overview pages.


## Pre-merge Checklist
